config/src/color.rs

```diff
@@ -42,6 +42,9 @@ pub struct Palette {
     /// A list of 8 colors corresponding to bright versions of the
     /// ANSI palette
     pub brights: Option<[RgbColor; 8]>,
+    /// A map for setting arbitrary colors ranging from 16 to 256 in the color
+    /// palette
+    pub indexed: Option<HashMap<String, RgbColor>>,
```

Comment:
This should probably be `u8` as we can only accommodate numbers 16-255

```suggestion
    pub indexed: Option<HashMap<u8, RgbColor>>,
```

Reply:
> This should probably be `u8` as we can only accommodate numbers 16-255

This would indeed be nicer, but then I get the runtime error `invalid type: string "66", expected u8`. Seems like the values of maps are automatically converted to strings. I think changing this behavior would require to modify [this](https://github.com/wez/wezterm/blob/9a112e82ac4079c3b344bb473d806f029fa58c79/luahelper/src/serde_lua/mod.rs#L225-L333) function. Before I try my hands on this, can you confirm that this assumption is correct?

Reply:
Ah, that might be too big a thing to bite off over there.
An alternative approach might be something like:

```rust
#[serde(default, deserialize_with = "de_indexed")]
pub indexed: HashMap<u8, RgbColor>,
```

Note: for a container type like `HashMap`, we can use `serde(default)` to allow defaulting to an empty map when the value isn't specified in the config, and that
simplifies the logic a bit when we consume the value.

Then we provide a custom deserializer that deserializes first into a type with
the string keys, and then transforms the value to match the type we want:

```rust
fn de_indexed<'de, D>(deserializer: D) -> Result<HashMap<u8,RgbColor>, D::Error>
where
    D: Deserializer<'de>,
{
  #[derive(Deserialize)]
  struct Wrap(HashMap<String, RgbColor>);
  let map = Wrap::deserialize(deserializer)?;

  Ok(map.into_iter().filter_map(|(k, v)| {
    match k.parse::<u8>() {
         Ok(n) if n >= 16 => Some((n, v))
         Err(_) => {
              log::warn!("Ignoring invalid color key {}", k);
              None
         }
    }
  }).collect())
}
```